src/D2L.Security.OAuth2/Keys/IPrivateKeyDataProvider.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace D2L.Security.OAuth2.Keys {
+	/// <summary>
+	/// An externally-implemented storage for private keys.
+	/// </summary>
+	public interface IPrivateKeyDataProvider {
+		/// <summary>
+		/// Save a private key.
+		/// </summary>
+		Task SaveAsync( PrivateKeyData keyData );

```

Comment:
We talked about not including this. It's back because when playing with where keygen should go it's nice if `KeyManagementService` is responsible for both generating and storing the key (to both places.)

We could theoretically have one `IKeyDataProvider` interface rather than the separation, but I think that'd be something to consider if we totally redesigned the library.